### PR TITLE
Keep adaptive sidebar compaction scoped to Browser

### PR DESCRIPTION
**Pull request by aprv10**
## What

Limit automatic sidebar-to-icon-rail compaction to the docked Browser workspace.

Summary, Reviews, and Files now retain the normal sidebar unless the user manually collapses it. Leaving Browser clears its workspace pressure immediately and restores the surrounding layout.

## Why

All inspector views previously published adaptive workspace pressure. On narrower windows, opening utility views such as Summary or Reviews could unexpectedly compact the sidebar even though those views did not need the additional canvas space.

Browser is the co-working surface that benefits from reclaiming navigation width. Applying the same behavior elsewhere made the layout feel inconsistent and overly reactive.

## How

- Added one browser-specific workspace-pressure predicate shared by:
  - Inspector tab transitions
  - Inspector open/close actions
  - Steady-state workspace sizing
- Clear adaptive pressure immediately when moving from Browser to Summary, Reviews, or Files.
- Preserve the existing manual sidebar expansion override.
- Retain the compact rail’s macOS traffic-light clearance.
- Added regression coverage for every inspector view and the macOS compact-rail offset.

## Testing

- `npm test -- --run src/renderer/components/SessionView.test.tsx src/renderer/components/Sidebar.test.tsx src/renderer/components/TitlebarNav.test.tsx src/renderer/stores/ui-store.test.ts src/renderer/test/shell-new-session-shortcut.test.tsx`
  - 183 tests passed
- `npm test -- src/renderer`
  - 141 test files passed
  - 2,104 tests passed
- `npm run typecheck`
- `npm run typecheck:e2e`
- `npx vite build --config vite.renderer.config.ts`

## Checklist

- [x] Branched from `main`
- [x] One focused change
- [x] Follows `AGENTS.md` conventions and PR hygiene
- [x] Tests added/updated for user-visible behavior

### CODE REVIEW

**Comment by Vaibhaav-Tiwari** [comment]
can you attach screenshots?

**Comment by somewherelostt** [comment]
Thanks for contributing to Agent Orchestrator.

This PR is being picked up by the current external contributor on-call pair:

- @illegalcall
- @somewherelostt

If someone is already working on this, please continue as usual.
The on-call pair is added for visibility, tracking, and support, not to take over the work.
If you need help with review, direction, reproduction, or next steps, please tag @illegalcall and @somewherelostt here.

For faster context or live questions, you can also join the AO Discord.

Join the session here:
https://discord.gg/H6ZDcUXmq

Come by if you want to see what is being built, ask questions, or just hang around with the community.

File: frontend/src/renderer/components/TitlebarNav.tsx [hunk 1]
```diff
@@ -50,12 +50,14 @@ export function TitlebarNav({
...
   const toggleSidebar = useUiStore((state) => state.toggleSidebar);
   const isSidebarOpen = useUiStore(sidebarIsVisible);
+  const isSidebarCompact = useUiStore(sidebarIsCompact);
   const sidebarHasLayout = useUiStore(sidebarOccupiesLayout);
   const router = useRouter();
   const canGoBack = useCanGoBack();
   const canGoForward = useCanGoForward();
 
   if (!isMac && !isLinux) return null;
+  if (isSidebarCompact) return null;
```
Comment (neversettle17-101) [inline frontend/src/renderer/components/TitlebarNav.tsx:60]: When `sidebarIsCompact` is true this returns `null`, but the new compact-rail replacement in `Sidebar` only adds a `PanelLeft` toggle. That removes the shell back/forward buttons entirely for macOS/Linux pointer users whenever Browser pressure compacts the sidebar. Please keep the history arrows available in compact mode or move them into the icon rail too.

**Review by neversettle17-101 — COMMENTED** [review]
## What This PR Does

Limits adaptive sidebar behavior to the docked Browser workspace, switches the session/chat topbars from measured widths to flex sizing, and adds compact-rail clearance handling plus tests.

## Summary of Changes

- Restricts sidebar auto-collapse pressure to the Browser inspector view instead of all inspector surfaces.
- Changes sidebar toggling under Browser pressure so a manual expansion returns to the compact rail instead of fully hiding the sidebar.
- Removes measured topbar width state from the terminal/chat headers and replaces it with flex layout plus macOS compact-rail padding classes.
- Hides the fixed titlebar nav in compact mode and adds a compact-rail sidebar toggle button in the sidebar header.
- Adds tests around compact-rail layout, toggle behavior, and topbar rendering.

## Findings

- **Medium**: Compact rail removes the only app-history buttons on macOS/Linux.

## Regression Risk

Browser-pressure sessions can lose mouse-accessible shell navigation on macOS/Linux.

## Testing

Added component/store tests for the new compact-rail behavior. I did not run tests locally.

Needs changes.

**Comment by Vaibhaav-Tiwari** [comment]
Can we get this done ASAP?

**Comment by aprv10** [comment]
> Can we get this done ASAP?

It will probably get merged today. Just needs testing

**Review by neversettle17-101 — COMMENTED** [review]
## What This PR Does

This PR confines adaptive sidebar compaction to the docked Browser workspace and reworks the session/chat topbars so tab chrome stays stable while space is reclaimed. It also moves compact-rail navigation controls into the sidebar itself and updates responsive session chrome behavior for branch badges, shell tabs, and embedded topbar actions.

## Flow

- SessionView now publishes sidebar workspace demand only when the Browser inspector is docked and open.
- The shell reacts to that demand by switching between expanded sidebar and compact icon rail without mutating the durable sidebar preference.
- CenterPane and ChatWorkspace both rely on flex layout plus compact-state selectors to preserve tab chrome and titlebar clearance while the Browser rail is active.

## Summary of Changes

- Restricts workspace-pressure-driven sidebar compaction to the Browser inspector path in `SessionView` and updates sidebar toggle semantics in `ui-store`.
- Refactors terminal/chat topbars to use shared flex-based tab regions instead of measured widths, including new compact session chrome styling.
- Adds compact branch-badge behavior, shrinkable connected shell/reviewer tabs, and compact embedded topbar action styling.
- Hides the fixed macOS/Linux titlebar nav while the compact rail is active and surfaces back/forward controls inside the sidebar rail instead.
- Adjusts collapsed-sidebar footer/search layout and adds regression coverage across session view, chat workspace, sidebar, titlebar nav, topbar, and store behavior.

## Findings

- No clear regression found from the reviewed diff.

## Regression Risk

Low. The change is tightly scoped to frontend shell/session layout behavior, and the new store/session-view wiring keeps Browser-specific compaction separate from utility inspector views. Residual risk is mostly visual/responsive edge cases that are hard to prove without running the app.

## Testing

Tests were not run. The diff adds targeted automated coverage in `CenterPane.test.tsx`, `SessionView.test.tsx`, `ChatWorkspace.test.tsx`, `Sidebar.test.tsx`, `ShellTopbar.test.tsx`, `TitlebarNav.test.tsx`, `SessionBranchBadge.test.tsx`, `ui-store.test.ts`, and `shell-new-session-shortcut.test.tsx`, which collectively cover the Browser-only compaction path, compact-rail toggle semantics, compact titlebar behavior, and the updated tab/branch chrome behavior.

## Open Questions

- None from the reviewed context.

Verdict: approved / ready to merge.

**Review by neversettle17-101 — APPROVED** [review]
